apps/src/netsim/NetSimVisualization.js

```diff
@@ -432,46 +440,46 @@ NetSimVisualization.prototype.updateVizEntitiesOfType_ = function (
 /**
  * Call kill() on any vizentities that match the given type and don't map to
  * a NetSimEntity in the provided collection.
- * @param {function} vizEntityType
+ * @param {function} vizElementType
  * @param {Array.<NetSimEntity>} entityCollection
  * @private
  */
 NetSimVisualization.prototype.killVizEntitiesOfTypeMissingMatch_ = function (
-    vizEntityType, entityCollection) {
-  this.elements_.forEach(function (vizEntity) {
-    var isCorrectType = (vizEntity instanceof vizEntityType);
+    vizElementType, entityCollection) {
+  this.elements_.forEach(function (vizElement) {
+    var isCorrectType = (vizElement instanceof vizElementType);
     var foundMatch = entityCollection.some(function (entity) {
-      return entity.entityID === vizEntity.id;
+      return vizElement.getCorrespondingEntityID &&
+          entity.entityID === vizElement.getCorrespondingEntityID();
```

Comment:
Interesting, sorta like an interface check :). Think it's fine the way it is, one potential approach I'm not sure would be better could be a comparison method that's on every `VizElement`, e.g. `correspondsToEntity(entityID)`, have the top-level one `return false`, then here and other comparison usages would be `vizElement.correspondsToEntityID(entity.entityID)`.

Though that kind of leaks the `entity`-tying concept up to `VizElement`, now thinking the way it is now might be preferable